pgx/src/bgworkers.rs

```diff
@@ -200,12 +203,12 @@ impl BackgroundWorker {
             pg_sys::StartTransactionCommand();
             pg_sys::PushActiveSnapshot(pg_sys::GetTransactionSnapshot());
         }
-        let result = pg_sys::guard(|| transaction_body());
         unsafe {
+            let result = pg_sys::guard(|| transaction_body());
```

Comment:
Is this right? I think it should be something like `PgxTryBuilder::new(|| transaction_body()).execute()` (possibly via a `pg_try` helper). `guard` pretty much should only be used in `extern "C"` fn`s, I think.

Reply:
hmm.  you're right.